### PR TITLE
Add developer documentation

### PR DESCRIPTION
**Pull request by rgreinho**
Adds a documentation page explaining how to setup a developer
environment.

Signed-off-by: Rémy Greinhofer <remy.greinhofer@gmail.com>

### CODE REVIEW

File: SETUP.md [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+## Prepare the working directories
+
+The directories are expected to be located in the same folder, as they depend on
+each other. These environment variables are provided to simplify the setup
+process. If you do not want to use them as is, you will need to adjust them
+according to your case.
+
+```bash
+# Export the variable(s).
+export GUACSEC_HOME="${HOME}/projects/guacsec"
```
Comment (lumjjb) [inline SETUP.md:17]: i usually use `go env GOPATH`/src/github.com/gaucsec, not sure if this is still the common convention

File: SETUP.md [hunk 2]
```diff
@@ -0,0 +1,54 @@
+# Setup
```
Comment (lumjjb) [inline SETUP.md:1]: nit: let's link to this via README.md and CONTRIBUTING.md

**Review by lumjjb — COMMENTED** [review]
Awesome!!! this is super helpful and will let folks try it out real easy!!

File: SETUP.md (hunk 1 above)
Comment (rgreinho) [reply SETUP.md:17]: This was the pre go-module convention. However, as I mentioned in the text above, people can adjust the value to whatever their like. It is mainly to simplify the instructions after  that.

File: SETUP.md [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+```bash
+cd guac
+make build
+```
+
+and use the `guacone` client:
+
+```bash
+bin/guacone files --creds neo4j:s3cr3t ${GUACSEC_HOME}/guac-data/top-dh-sboms
+```
```
Comment (lumjjb) [inline SETUP.md:54]: minor: hmm, should we add a step here to tell people that they can browse the neo4j graph at `http://localhost:7474/`?

File: SETUP.md [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+cd ${GUACSEC_HOME}
+
+# Clone the Guacsec repositories.
+git clone git@github.com:guacsec/guac-data.git
+git clone git@github.com:guacsec/guac.git
+```
+
+## Setup Neo4j
+
+Guac uses Neo4j as a graph database. Neo4j provides and maintain official Docker
```
Comment (lumjjb) [inline SETUP.md:30]:
```suggestion
GUAC uses Neo4j as a graph database. Neo4j provides and maintain official Docker
```

File: SETUP.md [hunk 5]
```diff
@@ -0,0 +1,54 @@
+# Setup
+
+## Requirements
+
+- Docker
+- Golang 1.19.2+
```
Comment (lumjjb) [inline SETUP.md:6]: we only use 1.18 in our go.mod so i think let's put 1.18?

Comment (mlieberman85) [reply SETUP.md:6]: +1

File: SETUP.md (hunk 3 above)
Comment (mlieberman85) [reply SETUP.md:54]: As well as being able to use any other neo4j client like the desktop

File: SETUP.md (hunk 5 above)
Comment (rgreinho) [reply SETUP.md:6]: Addressed ✅

File: SETUP.md (hunk 4 above)
Comment (rgreinho) [reply SETUP.md:30]: Addressed ✅

File: SETUP.md (hunk 3 above)
Comment (rgreinho) [reply SETUP.md:54]: Addressed ✅

**Comment by mlieberman85** [comment]
Minor nit, and then I think it's good.

File: Makefile [hunk 6]
```diff
@@ -49,3 +49,9 @@ clean: ## Remove temporary files
 help:
 	@grep -E '^[a-zA-Z_-]+:.*?## .*$$' $(MAKEFILE_LIST) | awk 'BEGIN {FS = ":.*?## "}; {printf "\033[36m%-30s\033[0m %s\n", $$1, $$2}'
 
+
+.PHONY:format fmt-md
+format: fmt-md ##  Run all the formatting tasks
+fmt-md: ## Format all the markdown files
+	npx --yes prettier --write --prose-wrap always **/*.md
```
[the remark this answers is not in the document]
Comment (rgreinho) [reply Makefile:56]: Addressed ✅

**Review by pxp928 — APPROVED** [review]
LGTM

**Review by lumjjb — APPROVED** [review]

**Pull request merged**